test/e2e/actions/extra_related.go

```diff
@@ -0,0 +1,42 @@
+package actions
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/customresource"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/actions/deploy"
+	kubecli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli/kubecli"
+	mongocli "github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/cli/mongocli"
+	"github.com/mongodb/mongodb-atlas-kubernetes/test/e2e/model"
+)
+
+const (
+	atlasClusterCRD = "atlasdeployments.atlas.mongodb.com"
+)
+
+func DeleteClusterCRD(data *model.TestDataProvider) {
+	By(fmt.Sprintf("Deleting %s", atlasClusterCRD), func() {
+		kubecli.DeleteClusterResource("crd", atlasClusterCRD)
+
+		By("Checking Cluster still existed", func() {
+			state := mongocli.GetClusterStateName(data.Resources.ProjectID, data.Resources.Clusters[0].Spec.DeploymentSpec.Name)
+			Expect(state).ShouldNot(Equal("DELETING"), "Cluster is being deleted despite the keep annotation")
+		})
+
+	})
+
+	// Remove annotation so actions.AfterEachFinalCleanup can cleanup successfully
```

Comment:
1. First we create a deployment with keep annotation
2. Delete cluster crd 
3. Check if the cluster still exists? (if no we exit the test with an error)
4. Cluster exists, for cleanup phase, we cannot execute `AfterEachFinalCleanup` successfully because there is no atlasDeployment CRD, and the cluster has `keep annotation`. 
5. We deployed CRD to ns again and remove `keep annotation` 

But I think i must move line 31-39 logic to somewhere outside of ` DeleteClusterCRD`

Reply:
```mermaid
graph LR
a[create `AtlasDeployment` with keep annotatoin] --> b[delete `AtlasDeployment`]
b --> d[does cluster exist in Atlas?]
d --> fail
d --> success
success --> cleanup[redeploy `AtlasDeployment` so cleanup phase succeeds]
```